presto-bigquery/src/main/java/com/facebook/presto/plugin/bigquery/BigQueryMetadata.java

```diff
@@ -157,7 +157,7 @@ public ConnectorTableLayout getTableLayout(ConnectorSession session, ConnectorTa
     private Optional<TableInfo> getBigQueryTable(SchemaTableName tableName)
     {
         TableInfo tableInfo = bigQueryClient.getTable(TableId.of(projectId, tableName.getSchemaName(), tableName.getTableName()));
-        return tableInfo == null ? Optional.of(tableInfo) : Optional.empty();
+        return Optional.ofNullable(tableInfo);
```

Comment:
This change looks good, were you able to test this change? Were you able to successfully query?

I was glancing through this class and I think a couple of more changes would be required. If possible, can you take a look?

https://github.com/prestodb/presto/pull/16201/files#diff-8632083d2d932d9a42410e2ccaa167bd6b0e436d58e01dbe653adb4194064dbfR232

https://github.com/prestodb/presto/pull/16201/files#diff-8632083d2d932d9a42410e2ccaa167bd6b0e436d58e01dbe653adb4194064dbfR119

Reply:
Ah good catch. I just noticed the obvious NPE when trying to box the optional with a null. I went ahead and I adjusted the other parts of the code.

I don't have big query set up to test the changes though. This diff is open to anyone wanting to take ownership of this diff to see it though to the end.

Reply:
@henneberger thanks for looking into this. There is another PR that has been raised with a couple of other fixes as well including these (16258).

Reply:
Great! Closing PR.